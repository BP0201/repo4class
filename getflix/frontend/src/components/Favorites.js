import React, { useEffect, useContext, useState } from 'react';
import MovieAPI from '../apis/MovieApi';
import UserContext from '../UserContext';
import DataCard from './DataCard';

export default function Favorites({ add, remove }) {
  const { userFavorites } = useContext(UserContext);
  const [resolvedFavorites, setResolvedFavorites] = useState([]);

  useEffect(function loadFavoritesOnMount() {
    getFavorites();
  }, []);

  async function getFavorites() {
    // create promise array of each favorite
    const favsInfo = userFavorites.map(f => MovieAPI.fetchSingle(f.id, f.id_type));
    // resolve promise array and set state
    setResolvedFavorites(await Promise.all(favsInfo));
    }

  // render all favorites as DataCards
  function handleFavorites(data) {
    return data.map(d => (
      <DataCard
        name={d.title || d.name}
        key={d.id}
        id={d.id}
        poster={ d.poster_path }
        release_date={d.release_date || d.first_air_date}
        // genres={d.genres}
        description={d.overview}
        category={d.first_air_date ? "tv" : "movie"}
        add={add}
        remove={remove}
      />
    ))}

  return (
    <div className='Favorites container'>
      <h1 className='display-4 m-3'>Favorites</h1>
      {resolvedFavorites.length
            ? (
                <div className="Favorites-DataCards">
                  {handleFavorites(resolvedFavorites)}
                </div>
            ) : (
                <p className="lead">You currently have no favorites.</p>
            )}
    </div>
  )
}