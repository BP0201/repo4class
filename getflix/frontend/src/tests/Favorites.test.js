import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Favorites from '../components/Favorites';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from './testUtils';

describe("Favorites", () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Favorites />
        </UserProvider>
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
  });
});