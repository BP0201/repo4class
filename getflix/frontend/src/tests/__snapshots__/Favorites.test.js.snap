// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Favorites container"
  >
    <h1
      class="display-4 m-3"
    >
      Favorites
    </h1>
    <p
      class="lead"
    >
      You currently have no favorites.
    </p>
  </div>
</DocumentFragment>
`;
